backend/volunteer_hub/views.py

```diff
@@ -1,21 +1,133 @@
+        if address_query:
+            volunteers = volunteers.filter(
+                address__icontains=address_query
+            ).distinct()
+
+        latitude = request.query_params.get('latitude')
+        longitude = request.query_params.get('longitude')
+        if latitude and longitude:
+            user_location = Point(float(longitude), float(latitude), srid=4326)
+            volunteers = volunteers.filter(
+                location__distance_lte=(user_location, D(km=10)))
+
         serializer = VolunteerSerializer(volunteers, many=True)
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class VolunteerTeamListView(APIView):
+    permission_classes = [AllowAny]
+
+    @swagger_auto_schema(
+        operation_description="Retrieve a list of volunteer Teams with optional filtering and searching.",
+        manual_parameters=[
+            openapi.Parameter(
+                'thana', openapi.IN_QUERY, description="Filter by Thana ID", type=openapi.TYPE_INTEGER
+            ),
+            openapi.Parameter(
+                'district', openapi.IN_QUERY, description="Filter by District ID", type=openapi.TYPE_INTEGER
+            ),
+            openapi.Parameter(
+                'address', openapi.IN_QUERY, description="Full-text search on volunteer addresses", type=openapi.TYPE_STRING
+            ),
+            openapi.Parameter(
+                'latitude', openapi.IN_QUERY, description="Latitude for geospatial search", type=openapi.TYPE_NUMBER
+            ),
+            openapi.Parameter(
+                'longitude', openapi.IN_QUERY, description="Longitude for geospatial search", type=openapi.TYPE_NUMBER
+            )
+        ],
+        responses={200: VolunteerTeamSerializer(many=True)},
+    )
+    def get(self, request, *args, **kwargs):
+        queryset = VolunteerTeam.objects.all()
+
+        # Filtering by district and thana using members' data
+        district_id = request.query_params.get('district')
+        thana_id = request.query_params.get('thana')
+        if district_id:
+            queryset = queryset.filter(
+                models.Q(members__district_id=district_id) |
+                models.Q(team_leader__district_id=district_id)
+            ).distinct()
+        if thana_id:
+            queryset = queryset.filter(
+                members__thana_id=thana_id
+            ).distinct()
+            queryset = queryset.filter(
+                models.Q(members__thana_id=thana_id) |
+                models.Q(team_leader__thana_id=thana_id)
+            ).distinct()
+
+        # Full-text search on address for both team_leader and members
+        address_query = request.query_params.get('address')
+        if address_query:
+            queryset = queryset.filter(
+                models.Q(team_leader__address__icontains=address_query) |
+                models.Q(members__address__icontains=address_query)
+            ).distinct()
```

Comment:
Filtering using addresses may exclude useful volunteers as people may not type addresses correctly. I would request you to give this a second thought.